src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -351,7 +351,7 @@ public static Expression<TDelegate> Lambda<TDelegate>(Expression body, String na
         public static Expression<TDelegate> Lambda<TDelegate>(Expression body, String name, bool tailCall, IEnumerable<ParameterExpression> parameters)
         {
             var parameterList = parameters.ToReadOnly();
-            ValidateLambdaArgs(typeof(TDelegate), ref body, parameterList);
+            ValidateLambdaArgs(typeof(TDelegate), ref body, parameterList, nameof(TDelegate));
```

Comment:
I'm not sure either. Indeed I'm not sure an argument exception is really appropriate here at all, though even if I'm correct it's probably not worth the compatibility break.